riscv-rt/macros/src/lib.rs

```diff
@@ -197,6 +204,7 @@ fn is_correct_type(ty: &Type, name: &str) -> bool {
 ///
 /// # fn main() {}
 /// ```
+#[deprecated(note = "Use global_asm! to define the __pre_init function instead")]
```

Comment:
Will we be removing this in the next (or another future) point release?

I wasn't aware that using Rust code before RAM init was unsound. Is this true even for code that only uses the stack?

Reply:
I'm not sure about how Rust's ABI can break something. I guess that, if you do not use static variables, it should be fine

Reply:
From `cortex-m` documentation:

> A `#[pre_init]` macro is also provided to run a function before RAM initialisation, but its use is deprecated as it is not defined behaviour to execute Rust code before initialisation. It is still possible to create a custom `pre_init` function using assembly.

Reply:
We also discussed this on Monday's meeting. I think the consensus was that there were too many preconditions + invariants to justify keeping a `pre_init` implementation in Rust.

I think it may still be helpful to point to the feature-gated `__pre_init` function in the `riscv-rt` docs as an example for users that would want to implement their own. Maybe in the place currently held by `#[pre_init]` or above the `_pre_init_trap`?